python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -342,28 +348,77 @@ def _messages_thread(
     ) -> None:
         try:
             start_or_last_download = datetime.datetime.now()
+            session_closed_at = None
+            cursor = None
+            can_read_messages = False
+
+            # main loop to read messages
+            # at every step, we:
+            # - exit early if we have received the closed message
+            # - consume params from the launched_payload if possible
+            # - check if we can start reading messages (e.g. log files are available)
+            # - download a chunk of messages and process them
+            # - if is_session_closed is set, we exit the loop after waiting for WAIT_FOR_LOGS_AFTER_EXECUTION_INTERVAL
             while True:
+                # if we have the closed message, we can exit
+                # since the message reader has been started and the external process has completed
+                if handler.received_closed_message:
+                    return
+
+                # always check for new params in case they have been updated
+                params = {**params, **(self.launched_payload or {})}
+
+                if not can_read_messages:
+                    can_read_messages = self.are_messages_readable(params)
```

Comment:
Oh I had an opposite impression that we agreed to make this more future-proof in case we are going to have multiple parameter updates... 

But I don't mind, it's also ok to do as you proposed.

Reply:
ok, let's go ahead and land